# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/FragmentOrElement.h
# Commit: c1e64e8e4665
# Full Hash: c1e64e8e466506d1d9a36d747df25527bd545b10
# Author: Martin Robinson <mrobinson@igalia.com>
# Date: 2022-12-01 16:18:29
# Regressor Bug: 1791759
# File Overlap Count: 1
# Description:
#   Bug 1791759 - Add support for `content-visibility: auto` r=emilio
#   
#   This change adds support for `content-visibilty: auto` as well as
#   showing and hiding content based on the relevancy of the content as
#   defined in the specification. Changes to relevancy are handled by
# ==============================================================================

diff -r 6d4d56f57b7b -r c1e64e8e4665 dom/base/FragmentOrElement.h
--- a/dom/base/FragmentOrElement.h	Thu Dec 01 01:23:08 2022 +0200
+++ b/dom/base/FragmentOrElement.h	Wed Nov 30 18:00:27 2022 +0000
@@ -14,6 +14,7 @@
 #define FragmentOrElement_h___
 
 #include "mozilla/Attributes.h"
+#include "mozilla/EnumSet.h"
 #include "mozilla/MemoryReporting.h"
 #include "mozilla/UniquePtr.h"
 #include "nsCycleCollectionParticipant.h"  // NS_DECL_CYCLE_*
@@ -33,6 +34,8 @@
 
 namespace mozilla {
 class DeclarationBlock;
+enum class ContentRelevancyReason;
+using ContentRelevancy = EnumSet<ContentRelevancyReason, uint8_t>;
 namespace dom {
 struct CustomElementData;
 class Element;
@@ -198,6 +201,18 @@
      */
     Maybe<float> mLastRememberedBSize;
     Maybe<float> mLastRememberedISize;
+
+    /**
+     * Whether the content of this element is relevant for the purposes
+     * of `content-visibility: auto.
+     */
+    Maybe<ContentRelevancy> mContentRelevancy;
+
+    /**
+     * Whether the content of this element is considered visible for
+     * the purposes of `content-visibility: auto.
+     */
+    Maybe<bool> mVisibleForContentVisibility;
   };
 
   class nsDOMSlots : public nsIContent::nsContentSlots {